### PR TITLE
258 review all external links and implement correct behaviour

### PR DESCRIPTION
**Pull request by gd2910**
## Summary of the changes

All IcLinks changed to reflect style rules regarding external links

## Related issue

#258

## Checklist

- [x] I have [manually accessibility tested](https://design.sis.gov.uk/accessibility/testing/manual-testing) any changes, if relevant.
- [x] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

**Review by GCHQ-Developer-530 — CHANGES_REQUESTED** [review]
Sorry this is going to be a massive faff 😭 but showIcon is a deprecated prop, so ic-link should show the external icon just by setting target to _blank. Please can you remove showIcon from each link? 😬

File: src/components/ComponentDetails/SlotTable/index.tsx [hunk 1]
```diff
@@ -62,7 +62,12 @@ const PropTable: React.FC<PropTableProps> = ({ slotData }) => {
       </ic-typography>
       <ic-typography apply-vertical-margins>
         <p>
-          <ic-link href="https://javascript.info/slots-composition">
+          <ic-link
+            href="https://javascript.info/slots-composition"
+            showIcon
+            target="_blank"
```
Comment (GCHQ-Developer-530) [inline src/components/ComponentDetails/SlotTable/index.tsx:68]:
```suggestion
            href="https://javascript.info/slots-composition"
            target="_blank"
```

**Review by GCHQ-Developer-530 — COMMENTED** [review]
Looks really good - just one showIcon left to remove then happy to approve

**Comment by GCHQ-Developer-741** [comment]
> 

Should showIcon be removed from the code examples in the ic-link page?

**Review by GCHQ-Developer-741 — APPROVED** [review]

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**